redaxo/src/core/lib/util/socket/socket.php

```diff
@@ -244,7 +261,28 @@ public function doRequest($method, $data = '')
         }
 
         $this->openConnection();
-        return $this->writeRequest($method, $this->path, $this->headers, $data);
+        $response = $this->writeRequest($method, $this->path, $this->headers, $data);
+
+        if ('GET' !== $method || !$this->followRedirects || !$response->isRedirection()) {
+            return $response;
+        }
+
+        $location = $response->getHeader('location');
+
+        if (!$location) {
+            return $response;
+        }
+
+        if (false === strpos($location, '//')) {
+            $socket = self::factory($this->host, $this->port, $this->ssl)->setPath($location);
+        } else {
+            $socket = self::factoryUrl($location);
+        }
+
+        $socket->setTimeout($this->timeout);
+        $socket->followRedirects($this->followRedirects - 1);
```

Comment:
Sollen die Header des Original-Requests hier auch noch übertragen werden? Vermutlich ja, oder?

Reply:
Ich weiß nicht wie das andere clients machen und ob es hier white/blacklists geben muss wg sicerheit,maber z.b. Auth-Basic o.ä. Würde man z.b. Erwarten dass es nochmal mitgeschickt wird

Reply:
Hab getestet, wie curl es macht. Da werden eigene Header erneut mitgeschickt (in meinem Fall `X-Foo: 1`).
Keine Ahnung, ob es irgendeine Blacklist gibt.
Ich werde hier dann aber alle gesetzten Header weiterleiten.